common/test/acceptance/tests/lms/test_bookmarks.py

```diff
@@ -276,24 +295,146 @@ def test_unreachable_bookmark(self):
 
     def test_page_size_limit(self):
         """
-        Scenario: We can get more bookmarks if page size is greater than default page size.
-        Note:
-            * Current Bookmarks API page_size value is 10.
-            * page_size value in bookmarks client side is set to 500.
+        Scenario: We can't get bookmarks more than default page size.
 
         Given that I am a registered user
         And I visit my courseware page
-        And I have bookmarked all the units available
+        And I have bookmarked all the 11 units available
         Then I click on Bookmarks button
         And I should see a bookmarked list
-        And bookmark list contains 11 bookmarked items
+        And bookmark list contains 10 bookmarked items
         """
         self._test_setup(11)
         self._bookmark_units(11)
 
         self.bookmarks_page.click_bookmarks_button()
         self.assertTrue(self.bookmarks_page.results_present())
-        self.assertEqual(self.bookmarks_page.count(), 11)
+        self.assertEqual(self.bookmarks_page.count(), 10)
+
+    def test_pagination_with_single_page(self):
+        """
+        Scenario: Bookmarks list pagination is working as expected for single page
+        Given that I am a registered user
+        And I visit my courseware page
+        And I have bookmarked all the 2 units available
+        Then I click on Bookmarks button
+        And I should see a bookmarked list with 2 bookmarked items
+        And I should see paging header and footer with correct data
+        And previous and next buttons are disabled
+        """
+        self._test_setup(num_chapters=2)
+        self._bookmark_units(num_units=2)
+
+        self.bookmarks_page.click_bookmarks_button()
+        self.assertTrue(self.bookmarks_page.results_present())
+        self._verify_pagination_info(
+            bookmark_count_on_current_page=2,
+            header_text='Showing 1-2 out of 2 total',
+            previous_button_enabled=False,
+            next_button_enabled=False,
+            current_page_number=1,
+            total_pages=1
+        )
+
+    def test_pagination_with_multiple_pages(self):
```

Comment:
This test is quite long. I think it'd be easier to read if it were to be split up into two tests: testing the next page button and testing the previous page button.